sdk/python/kfp/compiler/pipeline_spec_builder.py

```diff
@@ -339,13 +339,17 @@ def build_component_spec_for_task(
             continue
 
         # skip inputs not present, as a workaround to support optional inputs.
-        if input_name not in task.inputs:
+        if input_name not in task.inputs and input_spec.default is None:
             continue
 
         if type_utils.is_parameter_type(input_spec.type):
             component_spec.input_definitions.parameters[
                 input_name].parameter_type = type_utils.get_parameter_type(
                     input_spec.type)
+            if input_spec.default is not None:
```

Comment:
Not an issue with this PR, but I do realize this is not a universal solution for all default values. Because default value could be `None`:
e.g.:
```
@component
def my_component(some_input: Optional[str] = None):
    print(some_input)
```
The way we support this is to not passing the parameter at all (not showing up in component spec input definitions), so that it pick up the default value from the function.

Reply:
I see, does unset behave the same as setting it to None? If they are the same, then probably it's okay here?